packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -160,9 +160,11 @@ class DraggableScrollableController {
   void _attach(_DraggableScrollableSheetScrollController scrollController) {
     assert(_attachedController == null, 'Draggable scrollable controller is already attached to a sheet.');
     _attachedController = scrollController;
+    _attachedController!.extent.addListener(notifyListeners);
```

Comment:
Somehow you'll have to move the listener over if the extent gets replaced via `_replaceExtent`.

Reply:
(Please add a test that covers this case)

Reply:
Good catch! Fixed behaviors and added tests.